packages/orbit-components/src/Text/__tests__/index.test.tsx

```diff
@@ -1,59 +1,146 @@
 import * as React from "react";
-import { screen, render } from "@testing-library/react";
 
+import { screen, render } from "../../test-utils";
+import theme from "../../defaultTheme";
+import { TEXT_ALIGN } from "../../common/tailwind/textAlign";
+import type { Props } from "../types";
+import type { SIZE_OPTIONS, WEIGHT_OPTIONS } from "../consts";
 import Text from "..";
-import { SIZE_OPTIONS, TYPE_OPTIONS } from "../consts";
-import defaultTheme from "../../defaultTheme";
+
+const SIZE_STYLES: { [K in SIZE_OPTIONS]: Record<"font-size" | "line-height", string> } = {
+  small: {
+    "font-size": theme.orbit.fontSizeTextSmall,
+    "line-height": theme.orbit.lineHeightTextSmall,
+  },
+  normal: {
+    "font-size": theme.orbit.fontSizeTextNormal,
+    "line-height": theme.orbit.lineHeightTextNormal,
+  },
+  large: {
+    "font-size": theme.orbit.fontSizeTextLarge,
+    "line-height": theme.orbit.lineHeightTextLarge,
+  },
+  extraLarge: {
+    "font-size": theme.orbit.fontSizeTextExtraLarge,
+    "line-height": theme.orbit.lineHeightTextExtraLarge,
+  },
+};
+
+const WEIGHT_STYLES: { [K in WEIGHT_OPTIONS]: Record<"font-weight", string> } = {
+  normal: {
+    "font-weight": theme.orbit.fontWeightNormal,
+  },
+  bold: {
+    "font-weight": theme.orbit.fontWeightBold,
+  },
+  medium: {
+    "font-weight": theme.orbit.fontWeightMedium,
+  },
+};
 
 describe("Text", () => {
-  it("should have expected DOM output", () => {
-    const dataTest = "test";
-    const text = "Children text";
-    const type = TYPE_OPTIONS.PRIMARY;
-    const size = SIZE_OPTIONS.SMALL;
-    const id = "id";
+  const dataTest = "test";
+  const text = "Children text";
+  const id = "id";
 
+  it("should have expected DOM output", () => {
     render(
-      <Text dataTest={dataTest} type={type} size={size} id={id} spaceAfter="normal">
+      <Text dataTest={dataTest} id={id}>
         {text}
       </Text>,
     );
 
-    expect(screen.getByTestId(dataTest)).toHaveAttribute("id", id);
-    expect(screen.getByTestId(dataTest)).toBeInTheDocument();
+    const textElement = screen.getByTestId(dataTest);
+
+    expect(textElement.tagName).toBe("P");
+    expect(textElement).toHaveAttribute("id", id);
+    expect(textElement).toBeInTheDocument();
     expect(screen.getByText(text)).toBeInTheDocument();
+    expect(textElement).toHaveStyle({
+      margin: 0,
+      "font-family": theme.orbit.fontFamily,
+      "font-size": theme.orbit.fontSizeTextNormal,
+      "line-height": theme.orbit.lineHeightTextNormal,
+      "text-align": "left",
+      color: expect.stringContaining(theme.orbit.paletteInkDark),
```

Comment:
I still think nested expects is a bit strange thing, let's avoid such practices :)

Reply:
I can actually remove this expect, since it is being asserted in Cypress